### PR TITLE
[read/write set analysis] adding some useful tests

### PR DESCRIPTION
**Pull request by sblackshear**
Adding tests for:
- reading an address from a vector stored in a struct (credit Daniel Perez)
- reading from secondary indexes
- reading nested fields

### CODE REVIEW

File: language/move-prover/bytecode/tests/read_write_set/secondary_index.exp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  2: $t3 := read_ref($t2)
+  3: $t4 := borrow_global<SecondaryIndex::B>($t3)
+  4: $t5 := borrow_field<SecondaryIndex::B>.b_addr($t4)
+  5: $t6 := read_ref($t5)
+  6: return $t6
+}
+
+
+[variant baseline]
+fun SecondaryIndex::read_secondary_index_from_formal_interproc($t0|a_addr: address): address {
```
Comment (sblackshear) [inline language/move-prover/bytecode/tests/read_write_set/secondary_index.exp:101]: This summary is incorrect--should be the same as `read_secondary_index_from_global`

Comment (danhper) [reply language/move-prover/bytecode/tests/read_write_set/secondary_index.exp:101]: My latest implementation returns `Ret(0): Formal(0)/0x1::SecondaryIndex::B/b_addr`  which I think is correct

Comment (sblackshear) [reply language/move-prover/bytecode/tests/read_write_set/secondary_index.exp:101]: Ah yes, that's what it should be.

**Review by tzakian — DISMISSED** [review]

**Pull request closed without merging**